### PR TITLE
New search features and improvements

### PR DESCRIPTION
**Pull request by JamesComey**

### CODE REVIEW

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py:210]: Please don't mix `@app.route` arguments and `request.args`.
Use `@app.route` for a path (like /search or /projects/<id>) and request.args for actual arguments with sensible defaults, where applicable.

So please move `offset` and `count` here as well

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: Could you create this with a real dictionary instead of loading from a string, which will happen every time at runtime and can probably not be optimized by the compiler at all and is also error prone, if you want to make adjustments later.

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: see above

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: see above

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: see above

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: see above

File: backend/flask_api.py
Comment (Drakulix) [inline backend/flask_api.py]: see above

**Review by Drakulix — CHANGES_REQUESTED** [review]
This looks quite good, I have basically request two minor changes.

1. Please don't to arguments in `@app.route` and please don't mix them with `request.args`

2. Please use python dict and array literals instead of parsing json for static query parts. That way the syntax can be checked at compile time and more optimizations are possible.

Once that is fixed, I will happily merge it. Good job.

File: backend/flask_api.py
Comment (JamesComey) [reply backend/flask_api.py]: Yeah I am all booked out today and can do it earliest tomorrow evening.

Comment (JamesComey) [reply backend/flask_api.py]: Okay I pushed the changes now.

Comment (JamesComey) [reply backend/flask_api.py]: should be all good now

**Review by Drakulix — DISMISSED** [review]
You could have used the dictionary literal, but this is fine as well. I will merge sometime today, once I rebased this on current master.

**Comment by Drakulix** [comment]
@JamesComey I could not help myself, but clean that up. Could please verify that this is still doing what it should. Also this needs some tests, then I could have merged that myself. Please make this your next task.

File: backend/flask_api.py
Comment (henderkes) [inline backend/flask_api.py:218]: I think this should be ^2

**Comment by henderkes** [comment]
search_simple and search_advanced seem to be 95% the same, could look into summing that up
looks nice otherwise, it's too late to test right now but I will tomorrow if needed

**Review by Drakulix — APPROVED** [review]
@JamesComey confirmed function

**Pull request merged**
